zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/utils/CamundaExporterITInvocationProvider.java

```diff
@@ -0,0 +1,168 @@
+import org.opensearch.testcontainers.OpensearchContainer;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+
+public class CamundaExporterITInvocationProvider
+    implements TestTemplateInvocationContextProvider, AfterAllCallback, BeforeAllCallback {
+
+  public static final String CONFIG_PREFIX = "camunda-record";
+  private final ElasticsearchContainer elsContainer =
+      TestSupport.createDefeaultElasticsearchContainer();
+  private final OpensearchContainer<?> osContainer = TestSupport.createDefaultOpensearchContainer();
+  private ElasticsearchClient elsClient;
+  private OpenSearchClient osClient;
+  private SearchClientAdapter elsClientAdapter;
+  private SearchClientAdapter osClientAdapter;
+
+  private final List<AutoCloseable> closeables = new ArrayList<>();
+
+  private ExporterConfiguration getConfigWithConnectionDetails(
+      final ConnectionTypes connectionType) {
+    final var config = new ExporterConfiguration();
+    config.getIndex().setPrefix(CONFIG_PREFIX);
+    config.getBulk().setSize(1); // force flushing on the first record
+    if (connectionType == ELASTICSEARCH) {
+      config.getConnect().setUrl(elsContainer.getHttpHostAddress());
+    } else if (connectionType == OPENSEARCH) {
+      config.getConnect().setUrl(osContainer.getHttpHostAddress());
+    }
+    config.getConnect().setType(connectionType.getType());
+    return config;
+  }
+
+  @Override
+  public void beforeAll(final ExtensionContext context) {
+    elsContainer.start();
+    osContainer.start();
+
+    final var osConfig = getConfigWithConnectionDetails(OPENSEARCH);
+    osClient = new OpensearchConnector(osConfig.getConnect()).createClient();
+    osClientAdapter = new SearchClientAdapter(osClient);
+
+    final var elsConfig = getConfigWithConnectionDetails(ELASTICSEARCH);
+    elsClient = new ElasticsearchConnector(elsConfig.getConnect()).createClient();
+    elsClientAdapter = new SearchClientAdapter(elsClient);
+
+    closeables.add(elsContainer);
+    closeables.add(osContainer);
+  }
+
+  @Override
+  public boolean supportsTestTemplate(final ExtensionContext extensionContext) {
+    // we only want to template tests in the class which ask for configuration and the client
+    // adapter.
+    final var testParams = extensionContext.getTestMethod().orElseThrow().getParameters();
+    if (testParams.length != 2) {
+      return false;
+    }
+
+    return (testParams[0].getType().equals(ExporterConfiguration.class)
+        && testParams[1].getType().equals(SearchClientAdapter.class));
+  }
+
+  @Override
+  public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(
+      final ExtensionContext extensionContext) {
+
+    try {
+      elsClient.indices().delete(req -> req.index("*"));
+      elsClient.indices().deleteIndexTemplate(req -> req.name("*"));
+      osClient.indices().delete(req -> req.index("*"));
+      osClient.indices().deleteIndexTemplate(req -> req.name("*"));
```

Comment:
🔧 this is not consistent with the change introduced in SchemaManagerIt

Reply:
there's not an easy way to know which testcase we are in unless we add a parameter that is not used for tests. This fundamentally accomplishes the same thing. For each test that modifies els we get 1 els wipe instead of 2.

Reply:
in general, I don't think the `InvocationContextProvider` is the right/best place to reset/refresh the clients. This is better suited in the `beforeEach`.

in the `beforeEach` (similarly to the `SchemaManagerIT`) you can access the `TestInfo`, which has knowledge in this case about `config` parameter. You can use that to know which connection type is configured, and clean the indices accordingly

Reply:
I don't think it can be done in the `beforeEach` for the specific test clases as the `TestInfo` doesn't have the parameter instance, just information about it. 

I think it's fine to do it in the invocation providers `AfterEach` block

Reply:
will the `AfterEach` run if an exception was thrown from one method run?